src/parser/h1_parser.py

```
@@ -67,6 +68,150 @@ class H1Parser:
+    @classmethod
+    def _require_flag_span(cls, field_name: str, value: object, width: int) -> None:
+        # 公式の初期値は 0 だが、提供データは未設定位置を空白で送ってくる（本リポジトリ
+        # の H1/H6 返還エリアの無損失 pin が実測済み）。空白は提供値として保持し、
+        # それ以外の文字と桁落ちは拒否する。
+        if not isinstance(value, str) or len(value) != width or set(value) - {"0", "1", " "}:
+            raise ValueError(f"H1 {field_name} must be {width} positional 0/1 refund flags")
+
+    @classmethod
+    def _require_vote_total(cls, field_name: str, value: object) -> None:
+        if isinstance(value, str) and not value.strip():
+            # 提供データは合計エリアを空白で送ることがある（parse 後は空文字）。
```

Ruff (0.16.1)

[warning] Comment contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF003)